llm_app/model_wrappers/huggingface_wrapper/pipelines.py

```diff
@@ -3,9 +3,44 @@
 
 
 class HFPipelineTask(BaseModel):
-    def __init__(self, model_name, device="cpu", **kwargs):
+    def __init__(self, model, device="cpu", **kwargs):
+        """
+    A wrapper class for Hugging Face's `Pipeline` class.
+
+    The `pipeline` function from Hugging Face is a utility factory method that creates 
+    a Pipeline to handle different tasks.
+    It supports tasks like text classification, translation, summarization, and many more. 
+
+    This wrapper class simplifies the process of initializing the pipeline and allows the user 
+    to easily change the underlying
+    model used for computations.
+
+    Parameters:
+    -----------
+    model : str, required
+        The model identifier from Hugging Face's model hub.
+    device : str, default='cpu'
+        The device where the computations will be performed. 
+        Supports 'cpu' or 'gpu'. Default is 'cpu'.
+    **kwargs : optional
+        Additional arguments form HF. 
+        Please check out https://huggingface.co/docs/transformers/main/main_classes/pipelines
+        for more information on the models and available arguments.
+
+    Attributes:
+    -----------
+    pipeline : transformers.Pipeline
+        The Hugging Face pipeline object.
+    tokenizer : transformers.PreTrainedTokenizer
+        The tokenizer associated with the pipeline.
+
+    Example:
+    --------
+    >>> pipe = HFPipelineTask('gpt2')
+    >>> result = pipe('Hello world')
+    """
         super().__init__(**kwargs)
-        self.pipeline = pipeline(model=model_name, device=device)
+        self.pipeline = pipeline(model=model, device=device)
```

Comment:
Are you sure we want to call it `model`? It usually means a callable model, not its name.
If we still want to have `model` instead of `model_name`, we should also change that in the derived classes.

Reply:
Yes because both HF and OpenAI use the same argument 'model' for the locator

Reply:
Ok. Then let's rename `model_name` -> `model` in `HFTextGenerationTask` and `HFFeatureExtractionTask`.